Makefile

```diff
@@ -15,123 +15,67 @@ K_BIN := $(K_RELEASE)/bin
 
-wasm_files=test.k wasm.k data.k kwasm-lemmas.k
-wasm_source_files:=$(patsubst %, $(wasm_submodule)/%, $(patsubst %.k, %.md, $(wasm_files)))
-eei_files:=eei.k
-eei_source_files:=$(patsubst %, $(eei_submodule)/%, $(patsubst %.k, %.md, $(eei_files)))
-ewasm_files:=ewasm-test.k driver.k ewasm.k kewasm-lemmas.k
-all_k_files:=$(ewasm_files) $(wasm_files) $(eei_files)
-
-deps: $(wasm_submodule)/make.timestamp $(eei_submodule)/make.timestamp definition-deps
-
-definition-deps: wasm-definitions eei-definitions
-
-wasm-definitions:
-	$(wasm_make) -B defn-llvm
-	$(wasm_make) -B defn-haskell
+EEI_FILES:=eei.md
+EEI_SOURCE_FILES:=$(patsubst %, $(eei_submodule)/%, $(EEI_FILES))
+EWASM_FILES:=ewasm-test.md driver.md ewasm.md kewasm-lemmas.md
+EWASM_SOURCE_FILES:=$(EWASM_FILES)
 
-eei-definitions: $(eei_source_files)
-	$(eei_make) -B defn-llvm
-	$(eei_make) -B defn-haskell
-
-$(wasm_submodule)/make.timestamp: $(wasm_source_files)
-	git submodule update --init --recursive
-	$(wasm_make) deps
-	touch $(wasm_submodule)/make.timestamp
-
-$(eei_submodule)/make.timestamp: $(eei_source_files)
-	git submodule update --init --recursive
-	touch $(eei_submodule)/make.timestamp
+deps:
+	$(KWASM_MAKE) deps
 
 # Building Definition
 # -------------------
 
-llvm_dir:=$(defn_dir)/llvm
-llvm_defn:=$(patsubst %, $(llvm_dir)/%, $(all_k_files))
-llvm_kompiled:=$(llvm_dir)/ewasm-test-kompiled/interpreter
-
-haskell_dir:=$(defn_dir)/haskell
-haskell_defn:=$(patsubst %, $(haskell_dir)/%, $(all_k_files))
-haskell_kompiled:=$(haskell_dir)/ewasm-test-kompiled/definition.kore
-
-main_module=EWASM-TEST
-syntax_module=EWASM-TEST-SYNTAX
-
-# Tangle definition from *.md files
-
-defn: defn-llvm defn-haskell
-defn-llvm: $(llvm_defn)
-defn-haskell: $(haskell_defn)
-
-$(llvm_dir)/%.k: %.md $(tangler)
-	@echo "==  tangle: $@"
-	mkdir -p $(dir $@)
-	pandoc --from markdown --to $(tangler) --metadata=code:.k $< > $@
-
-$(haskell_dir)/%.k: %.md $(tangler)
-	@echo "==  tangle: $@"
-	mkdir -p $(dir $@)
-	pandoc --from markdown --to $(tangler) --metadata=code:.k $< > $@
+MAIN_MODULE=EWASM-TEST
+MAIN_SYNTAX_MODULE=EWASM-TEST-SYNTAX
+MAIN_DEFN_FILE=ewasm-test
 
 # Build definitions
 
 KOMPILE_OPTS :=
 
 build: build-llvm build-haskell
-build-llvm: $(llvm_kompiled)
-build-haskell: $(haskell_kompiled)
-
-$(llvm_kompiled): $(llvm_defn)
-	@echo "== kompile: $@"
-	kompile --backend llvm                     \
-	    --directory $(llvm_dir) -I $(llvm_dir) \
-	    --main-module $(main_module)           \
-	    --syntax-module $(syntax_module) $<    \
-	    $(KOMPILE_OPTS)
-
-$(haskell_kompiled): $(haskell_defn)
-	@echo "== kompile: $@"
-	kompile --backend haskell                        \
-	    --directory $(haskell_dir) -I $(haskell_dir) \
-	    --main-module   $(main_module)               \
-	    --syntax-module $(syntax_module) $<          \
-	    $(KOMPILE_OPTS)
+
+build-%:
+	cp $(EEI_SOURCE_FILES) $(EWASM_SOURCE_FILES) $(KWASM_SUBMODULE)
+	$(KWASM_MAKE) build-$*                               \
```

Comment:
You should have something more like:

```
build-%: $(EXTRA_DEFN_FILES)
    $(KWASM_MAKE) build-$* ...

$(KWASM_SUBMODULE)/%.md: %.md
    cp $< $@
```

so that `make` is able to correctly calculate that it should only copy the new files into place when they're updated, and then the call to `$(KWASM_MAKE)` will only actually rebuild the definition when the copied files into place are newer than the old build.

Reply:
Done.

This dependency check doesn't work when I add `build-llvm` and `build-haskell` as `.PHONY` targets. I guess because there _is_ no `build-llvm` or `build-haskell` target, only a `build-%` one. However, if I add `build-%` as a `.PHONY`, then the dependency checks work. Perhaps `.PHONY` assumes that any target that isn't defined with an exact name match doesn't exist.

Might be good to update that in the KWasm Makefile, too.